### ISSUE TITLE
Make metadata updateable and remove from claim hash

### ISSUE THREAD

**Issue by bbenligiray** [api3dao/claims-manager#66]
Currently, `metadata` is used in the policy hash, which means it always needs to be provided whenever a policy-related thing is to be done and it can't be updated. I think we can do something like [this](https://github.com/api3dao/api3-dao/blob/main/packages/convenience/contracts/Convenience.sol#L63), except it would emit the metadata as an event rather than storing it.

This means `CreatedClaim` won't emit `metadata`, but we can have a separate `UpdatedPolicyMetadata` event with indexed `claimant` to avoid having to make a `getLogs()` call per claim.

**Comment by mcoetzee** [comment]
Hmm just to be clear, `metadata` is used in the policy hash, and the policy hash is used in turn in the claim hash:
```solidity
bytes32 claimHash = keccak256(
    abi.encodePacked(
        policyHash,
        claimant,
        beneficiary,
        claimAmountInUsd,
        evidence
    )
);
```

So we don't provide the `metadata` for all the claim related functions, we provide it only for these 4 functions:
```
createPolicy()
upgradePolicy()
downgradePolicy()
createClaim()
```

**Comment by bbenligiray** [comment]
My bad, updated the OP

**Comment by mcoetzee** [comment]
My apologies, but I'm still not sure I understand the need to make the metadata updateable. I can't imagine that we would update the asset pair of the policy, because that sounds like a new policy to me rather than an update to an existing policy?


> ...it would emit the metadata as an event rather than storing it.

Lastly, `metadata` is currently an input to the policy hash, but I don't think that means that we store it though?

**Comment by bbenligiray** [comment]
> My apologies, but I'm still not sure I understand the need to make the metadata updateable

We can make a mistake while creating it, and we won't necessarily only have the asset pair in the metadata so that may need to be updated. This became more of a need with `upgradePolicy()` and policies potentially living for years. When I think about the beacon names in the operations repo for example, we changed their names multiple times.

> Lastly, metadata is currently an input to the policy hash, but I don't think that means that we store it though?

I mean in the link that I gave the discussion URL is stored, we wouldn't do the same here
